tools/camera/camera.go

```diff
@@ -87,30 +90,40 @@ func (c Camera) GetPresets() ([]model.CameraPreset, error) {
 //Example usage: c.makeAuthenticatedRequest("GET", "/base","/some.cgi?preset=1")
 //Returns the response body as a buffer.
 func (c Camera) makeAuthenticatedRequest(method string, body string, url string) (*bytes.Buffer, error) {
-	var camCurl *exec.Cmd
+	// var camCurl *exec.Cmd
+	userPassword := strings.Split(c.Auth, ":")
+	client := &http.Client{
+		Transport: &digest.Transport{
+			Username: userPassword[0],
+			Password: userPassword[1],
+		},
+	}
+
+	var req *http.Request
+	var err error
 	switch method {
 	case "GET":
-		camCurl = exec.Command("curl",
-			"--digest", "--user", c.Auth,
-			fmt.Sprintf("http://%s%s", c.Ip, url))
+		req, err = http.NewRequest("GET", fmt.Sprintf("http://%s%s", c.Ip, url), nil)
 	case "POST":
-		camCurl = exec.Command("curl",
-			"--digest", "--user", c.Auth,
-			"-d", body,
-			fmt.Sprintf("http://%s%s", c.Ip, url))
+		req, err = http.NewRequest("POST", fmt.Sprintf("http://%s%s", c.Ip, url), bytes.NewReader([]byte(body)))
 	default:
 		return nil, fmt.Errorf("unsupported protocol: %v", method)
 	}
-	var bts []byte
-	buf := bytes.NewBuffer(bts)
-	camCurl.Stdout = buf
-	err := camCurl.Start()
 	if err != nil {
+		log.WithError(err).Error("could not create http request")
```

Comment:
This module is intended to be somewhat importable, instead of logging errors we should return them with some extra context so they can be handled further up in the call chain:

```suggestion
		return nil, fmt.Errorf("create http request: %v", err)
```